rllib/core/learner/tests/test_learner.py

```diff
@@ -77,57 +59,51 @@ def test_compute_gradients(self):
         Tests that if we sum all the trainable variables the gradient of output w.r.t.
         the weights is all ones.
         """
-        learner = BCTfLearner(
-            module_spec=SingleAgentRLModuleSpec(
-                module_class=DiscreteBCTFModule,
-                observation_space=self.ENV.observation_space,
-                action_space=self.ENV.action_space,
-                model_config_dict={"fcnet_hiddens": [32]},
-            ),
-            # made this a configurable hparam to avoid information leakage in tests
-            # where we need to know what the learning rate is.
-            optimizer_config={"lr": 1e-3},
-            learner_group_scaling_config=LearnerGroupScalingConfig(),
-            framework_hyperparameters=FrameworkHyperparameters(eager_tracing=True),
-        )
-
-        learner.build()
-
-        with tf.GradientTape() as tape:
-            params = learner.module[DEFAULT_POLICY_ID].trainable_variables
-            loss = {ALL_MODULES: sum(tf.reduce_sum(param) for param in params)}
-            gradients = learner.compute_gradients(loss, tape)
-
-        # type should be a mapping from ParamRefs to gradients
-        self.assertIsInstance(gradients, dict)
-
-        for grad in gradients.values():
-            check(grad, np.ones(grad.shape))
+        for fw in framework_iterator(frameworks=("torch", "tf2")):
+            learner = get_learner(framework=fw, eager_tracing=True, env=self.ENV)
+
+            params = learner.get_parameters(learner.module[DEFAULT_POLICY_ID])
+
+            tape = None
+            if fw == "torch":
+                loss_per_module = {ALL_MODULES: sum(param.sum() for param in params)}
+            else:
+                with tf.GradientTape() as tape:
+                    loss_per_module = {
+                        ALL_MODULES: sum(tf.reduce_sum(param) for param in params)
+                    }
+
+            gradients = learner.compute_gradients(loss_per_module, gradient_tape=tape)
+
+            # Type should be a mapping from ParamRefs to gradients.
+            self.assertIsInstance(gradients, dict)
+
+            for grad in gradients.values():
+                check(grad, np.ones(grad.shape))
 
     def test_postprocess_gradients(self):
-        """Tests the postprocess_gradients correctness."""
-        config = (
-            APPOConfig()
-            .environment("CartPole-v1")
-            .framework(eager_tracing=True)
-            .rollouts(rollout_fragment_length=50)
-        )
-
-        # TODO (sven): Enable torch once available for APPO.
-        for _ in framework_iterator(config, frameworks=("tf2")):
-            # Clip by value only.
-            config.training(
-                grad_clip=0.75,
-                grad_clip_by="value",
+        """Tests the base grad clipping logic in `postprocess_gradients()`."""
```

Comment:
I think you added this test in one of the previous PRs. The problem I see with this test is that it is not a unittest for learner as it has a lot of dependency on the correct implementation of learner and learner_group and how they interact with the config. We do not want that kind of dependency in this file which is mean to be only testing the basics of Learner itself. Let's modify this test to use `get_learner()` utility function?

Reply:
The problem right now is that if any thing else bugs out in the algorithm stack this test would fail.

Reply:
Also can we somehow avoid the repeatition of similar logic in this code. The test is overly complicated and verbose.

Reply:
Simplified the test case:
- no longer AlgorithmConfig dependent
- less convoluted b/c uses a simpler LearnerHPs object directly.